### PR TITLE
Move return call to try/catch block

### LINKED ISSUE #9658: SecretClient.PurgeDeletedSecret is not properly recording failures
Seems to be a from a bad merge: https://github.com/Azure/azure-sdk-for-net/blob/13aca35b851f66d865e2a95026d1c5eadcfdc85e/sdk/keyvault/Azure.Security.KeyVault.Secrets/src/SecretClient.cs#L683

### PR DESCRIPTION
**Pull request by heaths**
Fixes #9658 which appears to be from a bad merge at some point.

### CODE REVIEW

File: sdk/keyvault/Azure.Security.KeyVault.Secrets/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Release History
 
+## 4.0.2 (Unreleased)
+
+### Minor changes
+
+- `SecretClient.PurgeDeletedSecret` properly traces errors ([#9658](https://github.com/Azure/azure-sdk-for-net/issues/9658))
```
Comment (maririos) [inline sdk/keyvault/Azure.Security.KeyVault.Secrets/CHANGELOG.md]: Could you explain to me when should we include a link to an issue vs just saying what the change was?

**Review by maririos — APPROVED** [review]

File: sdk/keyvault/Azure.Security.KeyVault.Secrets/CHANGELOG.md (hunk 1 above)
Comment (heaths) [reply sdk/keyvault/Azure.Security.KeyVault.Secrets/CHANGELOG.md]: I'm trying to figure that out. I included this as an example and asked a few people, though with no response. Will ping them. It's common in many other first- and third-party projects, though. PowerShell Core and VSCode, for example, do a great job of this. When the bugs are opened by customers, they really appreciate the recognition for finding and even fixing them. It fosters great community engagement.

**Review by mikeharder — APPROVED** [review]

File: sdk/keyvault/Azure.Security.KeyVault.Secrets/CHANGELOG.md (hunk 1 above)
Comment (maririos) [reply sdk/keyvault/Azure.Security.KeyVault.Secrets/CHANGELOG.md]: When you find an asnwer, could you share it with the team?
This is the first time I see it, and will be good that team wise we show consistency in our changelogs.

Comment (heaths) [reply sdk/keyvault/Azure.Security.KeyVault.Secrets/CHANGELOG.md]: @adrianhall what do you think about linking issues post-GA? See my response above for context.

Comment (heaths) [reply sdk/keyvault/Azure.Security.KeyVault.Secrets/CHANGELOG.md]: Talked with Adrian offline and he agrees this is acceptable. I will open an issue to update the guidance.

**Pull request merged**